batchflow/models/torch/base.py

```diff
@@ -648,50 +665,75 @@ def _make_optimizer(self, config):
+                raise ValueError('Missing `name` key in the decay configuration')
 
-        for decay, decay_args in res:
-            if decay is None:
-                return decays, list_kwargs, list_steps
+            # Parse decay
+            if callable(decay_) or isinstance(decay_, type):
+                pass
+            elif isinstance(decay_, str) and hasattr(torch.optim.lr_scheduler, decay_):
+                decay = getattr(torch.optim.lr_scheduler, decay_)
+            elif decay_ in DECAYS:
+                decay_ = DECAYS.get(decay_)
+            else:
+                raise ValueError('Unknown learning rate decay method', decay_)
 
-            step_meta_keys = ['frequency', 'first_iter', 'last_iter']
-            step_meta_defaults = [None, 0, np.inf]
-            step_meta = {key: decay_args.pop(key, default) for key, default in zip(step_meta_keys, step_meta_defaults)}
-            if step_meta['frequency'] is None:
+            # Parse step parameters
+            step_params = {
+                'first_iter': 0,
+                'last_iter': np.inf,
+                **decay_kwargs
+            }
+            if 'frequency' not in step_params:
                 raise ValueError('Missing `frequency` key in the decay configuration')
 
-            if callable(decay) or isinstance(decay, type):
-                pass
-            elif isinstance(decay, str) and hasattr(torch.optim.lr_scheduler, decay):
-                decay = getattr(torch.optim.lr_scheduler, decay)
-            elif decay in DECAYS:
-                decay = DECAYS.get(decay)
+            # Set defaults for some of the decays
+            if decay_ in DECAYS_DEFAULTS:
+                decay_dict = DECAYS_DEFAULTS.get(decay_).copy()
+                if decay == DECAYS['cos']:
+                    decay_dict.update(T_max=step_params['frequency'])
+                decay_kwargs = {**decay_dict, **decay_kwargs}
+
+            # Remove unnecessary keys from kwargs
+            for key in ['first_iter', 'last_iter', 'frequency']:
+                decay_kwargs.pop(key, None)
+
+            # Create decay or store parameters for later usage
+            if optimizer:
+                decay_ = decay_(optimizer, **decay_kwargs)
             else:
-                raise ValueError('Unknown learning rate decay method', decay)
+                decay = (decay_, decay_kwargs)
 
-            if decay in DECAYS_DEFAULTS:
-                decay_dict = DECAYS_DEFAULTS.get(decay).copy()
-                if decay == DECAYS['cos']:
-                    decay_dict.update(T_max=step_meta['frequency'])
-                decay_args = {**decay_dict, **decay_args}
+            self.decay.append(decay_)
+            self.decay_step.append(step_params)
 
-            decays.append(decay)
-            list_kwargs.append(decay_args)
-            list_steps.append(step_meta)
-        return decays, list_kwargs, list_steps
 
+    # Use external model
+    def set_model(self, model):
```

Comment:
model.set_model() is rather verbose and not obvious. Perhaps `use` or `with` or `create_with`

Reply:
The naming of `set_model` and `build_model` was used as a preparation for splitting the `TorchModel` class into two parts: one that creates `nn.Module` and one that trains it. I agree that it was too early to rename the `_build` method; can we keep the `set_model` name as it is most descriptive of what the method does?

Reply:
As far as I'm concerned, `set_model` is less descriptive.